tools/autograd/gen_python_functions.py

```diff
@@ -115,230 +107,217 @@ def should_generate_py_binding(f: NativeFunction) -> bool:
+def gen(out: str, native_yaml_path: str, deprecated_yaml_path: str, template_path: str) -> None:
+    fm = FileManager(install_dir=out, template_dir=template_path, dry_run=False)
 
-    env = create_python_bindings(py_torch_functions, is_python_method=False, module="torch")
+    methods = load_signatures(native_yaml_path, deprecated_yaml_path, method=True)
+    create_python_bindings(
+        fm, methods, is_py_variable_method, None, 'python_variable_methods.cpp', method=True)
 
-    write(out, 'python_torch_functions.cpp', PY_TORCH_FUNCTIONS_CPP, env)
+    functions = load_signatures(native_yaml_path, deprecated_yaml_path, method=False)
+    create_python_bindings(
+        fm, functions, is_py_torch_function, 'torch', 'python_torch_functions.cpp', method=False)
 
+    create_python_bindings(
+        fm, functions, is_py_nn_function, 'torch.nn', 'python_nn_functions.cpp', method=False)
 
-def group_declarations_by_op_name(declarations):
-    groups = defaultdict(list)
-    for d in declarations:
-        groups[op_name(d)].append(d)
-    return groups
+    create_python_bindings(
+        fm, functions, is_py_fft_function, 'torch.fft', 'python_fft_functions.cpp', method=False)
 
+    create_python_bindings(
+        fm, functions, is_py_linalg_function, 'torch.linalg', 'python_linalg_functions.cpp', method=False)
 
-def create_python_bindings(python_functions, is_python_method, module):
+def create_python_bindings(
+    fm: FileManager,
+    pairs: Sequence[PythonSignatureNativeFunctionPair],
+    pred: Callable[[NativeFunction], bool],
+    module: Optional[str],
+    filename: str,
+    *,
+    method: bool,
+) -> None:
     """Generates Python bindings to ATen functions"""
     py_methods: List[str] = []
     py_method_defs: List[str] = []
     py_forwards: List[str] = []
 
-    for name in sorted(python_functions.keys()):
-        overloads = list(decl_to_signature_function_pair(decl, method=is_python_method)
-                         for decl in python_functions[name])
-        py_methods.append(method_impl(name, module, overloads, method=is_python_method))
-        py_method_defs.append(method_def(name, module, overloads, method=is_python_method))
-        py_forwards.extend(forward_decls(name, overloads, method=is_python_method))
+    grouped: Dict[str, List[PythonSignatureNativeFunctionPair]] = defaultdict(list)
+    for pair in pairs:
+        if pred(pair.function):
+            grouped[str(pair.function.func.name.name)].append(pair)
+
+    for name in sorted(grouped.keys()):
+        overloads = grouped[name]
+        py_methods.append(method_impl(name, module, overloads, method=method))
+        py_method_defs.append(method_def(name, module, overloads, method=method))
+        py_forwards.extend(forward_decls(name, overloads, method=method))
 
-    return {
+    fm.write_with_template(filename, filename, lambda: {
+        'generated_comment': f'@generated from {fm.template_dir}/{filename}',
```

Comment:
Don't do this, now editors think this file is generated. '@' + 'generated' or something similar

Reply:
done